src/containers/SongsTable/SongsTable.js

```diff
@@ -12,29 +13,12 @@ class SongsTable extends Component {
   }
 
   componentDidMount() {
-    this.fetchData()
-  }
-
-  fetchData() {
-    const query = `{
-      Songs {
-        artist,
-        title,
-        image,
-        url
-      }
-    }`
-    fetch(`https://back.christmastop100.nl/graphql?query=${query}`)
-      .then(response => response.json())
-      .then(json => {
-        this.setState({
-          loaded: true,
-          data: json.data.Songs
-        })
-      })
-      .catch(error => {
-        console.log(error);
-      });
+    XmasAPI.fetchSongs(this, function (json, context) {context.setState({
+      loaded: true,
+      data: json.data.Songs
+    })}, function (error, context) {
+      console.log(error);
+    });
```

Comment:
Dit kan met Promises mooier opgelost worden. 
De fetchSongs static method zou een promise kunnen returnen i.p.v. een callback functie callen.
Hoef je verder ook geen context mee te sturen.
`XmasAPI.fetchSongs().then((data) => {
this.setState({
loaded: true,
data: data.Songs
})}`